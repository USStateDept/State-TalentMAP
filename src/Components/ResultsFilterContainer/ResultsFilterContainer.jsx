import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { FILTER_ITEMS_ARRAY, ACCORDION_SELECTION_OBJECT,
  MISSION_DETAILS_ARRAY, POST_DETAILS_ARRAY, USER_PROFILE } from '../../Constants/PropTypes';
import { ACCORDION_SELECTION } from '../../Constants/DefaultProps';
import SearchFiltersContainer from '../SearchFilters/SearchFiltersContainer/SearchFiltersContainer';
import ResetFilters from '../ResetFilters/ResetFilters';

class ResultsFilterContainer extends Component {
  constructor(props) {
    super(props);
    this.onQueryParamUpdate = this.onQueryParamUpdate.bind(this);
    this.onQueryParamToggle = this.onQueryParamToggle.bind(this);
  }

  shouldComponentUpdate(nextProps) {
    if (this.props !== nextProps) {
      return true;
    }
    return false;
  }

  onQueryParamUpdate(e) {
    this.props.onQueryParamUpdate(e);
    this.props.onChildToggle();
  }

  onQueryParamToggle(param, value, remove) {
    this.props.onQueryParamToggle(param, value, remove);
    this.props.onChildToggle();
  }

  render() {
    const { filters, resetFilters, setAccordion, selectedAccordion,
      fetchMissionAutocomplete, missionSearchResults, missionSearchIsLoading,
      missionSearchHasErrored, fetchPostAutocomplete, userProfile,
      postSearchResults, postSearchIsLoading, postSearchHasErrored } = this.props;
    return (
      <div className="filter-container">
        <div className="filter-container-bottom">
          <div className="usa-grid-full filter-control-container">
            <div className="filter-control-left">Select Filter:</div>
            <div className="filter-control-right">
              <ResetFilters resetFilters={resetFilters} />
            </div>
          </div>
          <div className="usa-grid-full search-filters-container">
            <SearchFiltersContainer
              filters={filters}
              queryParamUpdate={this.onQueryParamUpdate}
              queryParamToggle={this.onQueryParamToggle}
              selectedAccordion={selectedAccordion}
              setAccordion={setAccordion}
              fetchMissionAutocomplete={fetchMissionAutocomplete}
              missionSearchResults={missionSearchResults}
              missionSearchIsLoading={missionSearchIsLoading}
              missionSearchHasErrored={missionSearchHasErrored}
              fetchPostAutocomplete={fetchPostAutocomplete}
              postSearchResults={postSearchResults}
              postSearchIsLoading={postSearchIsLoading}
              postSearchHasErrored={postSearchHasErrored}
              isCDO={userProfile.is_cdo || false}
            />
          </div>
        </div>
      </div>
    );
  }
}

ResultsFilterContainer.propTypes = {
  filters: FILTER_ITEMS_ARRAY.isRequired,
  onQueryParamUpdate: PropTypes.func.isRequired,
  onChildToggle: PropTypes.func.isRequired,
  onQueryParamToggle: PropTypes.func.isRequired,
  resetFilters: PropTypes.func.isRequired,
  setAccordion: PropTypes.func.isRequired,
  selectedAccordion: ACCORDION_SELECTION_OBJECT,
  fetchMissionAutocomplete: PropTypes.func.isRequired,
  missionSearchResults: MISSION_DETAILS_ARRAY.isRequired,
  missionSearchIsLoading: PropTypes.bool.isRequired,
  missionSearchHasErrored: PropTypes.bool.isRequired,
  fetchPostAutocomplete: PropTypes.func.isRequired,
  postSearchResults: POST_DETAILS_ARRAY.isRequired,
  postSearchIsLoading: PropTypes.bool.isRequired,
  postSearchHasErrored: PropTypes.bool.isRequired,
  userProfile: USER_PROFILE.isRequired,
};

ResultsFilterContainer.defaultProps = {
  selectedAccordion: ACCORDION_SELECTION,
};

export default ResultsFilterContainer;
