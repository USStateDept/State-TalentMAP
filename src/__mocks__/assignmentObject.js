const assignmentObject = {
  start_date: '10-10-2016',
  end_date: '10-10-2018',
  estimated_end_date: '10-10-2018',
  position: {
    skill: 'CONSTRUCTION ENGINEERING (6218)',
    bureau: '(NEA) BUREAU NEAR EASTERN AFFAIRS',
    position_number: '58348065',
    title: 'PROJECT DIRECTOR',
    post: {
      location: {
        id: 4,
        country: 'United States',
        code: '500',
        city: 'Washington',
        state: 'DC',
      },
    },
  },
};

export default assignmentObject;
