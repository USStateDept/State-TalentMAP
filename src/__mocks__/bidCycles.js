export default [
  {
    id: 1,
    name: 'Demo BidCycle 2018-03-13 15:51:10.980576+00:00',
    cycle_start_date: '2017-12-13T15:51:10.980554Z',
    cycle_deadline_date: '2018-04-13T15:51:10.980554Z',
    cycle_end_date: '2018-06-13T15:51:10.980554Z',
    active: true,
  },
];
