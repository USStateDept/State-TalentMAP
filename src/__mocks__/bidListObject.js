const bidListObject = {
  count: 3,
  next: null,
  previous: null,
  results: [
    {
      id: 3,
      bidcycle: 'Demo BidCycle 2017-10-31 16:00:51.254762',
      user: 'rehmant',
      position: {
        id: 226,
        grade: '03',
        skill: 'POLITICAL AND ECONOMIC AFFAIRS (6050)',
        position_number: '10035561',
        title: 'POLITICAL/ECONOMIC OFFICER',
        create_date: '2015-08-05',
        update_date: '2017-06-08',
        bureau: '(AF) BUREAU OF AFRICAN AFFAIRS',
        post: {
          id: 199,
          location: {
            id: 103,
            country: 'Sierra Leone',
            code: '00A',
            city: 'Freetown',
            state: '',
          },
        },
      },
      status: 'approved',
      draft_date: '2017-12-20',
      submitted_date: '2017-12-20',
      handshake_offered_date: '2017-12-20',
      handshake_accepted_date: '2017-12-20',
      handshake_declined_date: '2017-12-20',
      in_panel_date: '2017-12-20',
      scheduled_panel_date: '2017-12-20',
      approved_date: '2017-12-20',
      declined_date: '2017-12-20',
      closed_date: '2017-12-20',
      create_date: '2017-12-20',
      update_date: '2017-12-20',
      is_priority: false,
      reviewer: {
        username: 'woodwardw',
        first_name: 'Wendy',
        last_name: 'Woodward',
        email: 'woodwardw@state.gov',
        phone_number: '555-555-5555',
        is_cdo: false,
      },
    },
    {
      id: 1,
      bidcycle: 'Demo BidCycle 2017-10-31 16:00:51.254762',
      user: 'rehmant',
      position: {
        id: 1,
        grade: '05',
        skill: 'OFFICE MANAGEMENT (9017)',
        position_number: '00003026',
        title: 'OMS (COM)',
        create_date: '2006-09-20',
        update_date: '2017-06-08',
        bureau: '(AF) BUREAU OF AFRICAN AFFAIRS',
        post: {
          id: 235,
          location: {
            id: 103,
            country: 'Sierra Leone',
            code: '00A',
            city: 'Freetown',
            state: '',
          },
        },
      },
      status: 'closed',
      draft_date: '2017-12-20',
      submitted_date: '2017-12-20',
      handshake_offered_date: '2017-12-20',
      handshake_accepted_date: '2017-12-20',
      handshake_declined_date: '2017-12-20',
      in_panel_date: '2017-12-20',
      scheduled_panel_date: '2017-12-20',
      approved_date: '2017-12-20',
      declined_date: '2017-12-20',
      closed_date: '2017-12-20',
      create_date: '2017-12-20',
      update_date: '2017-12-20',
      is_priority: true,
      reviewer: {
        username: 'woodwardw',
        first_name: 'Wendy',
        last_name: 'Woodward',
        email: 'woodwardw@state.gov',
        phone_number: '555-555-5555',
        is_cdo: false,
      },
    },
    {
      id: 2,
      bidcycle: 'Demo BidCycle 2017-10-31 16:00:51.254762',
      user: 'rehmant',
      position: {
        id: 84,
        grade: '04',
        skill: 'SECURITY (2501)',
        position_number: '56001600',
        title: 'SPECIAL AGENT',
        create_date: '2006-09-20',
        update_date: '2017-06-08',
        bureau: '(AF) BUREAU OF AFRICAN AFFAIRS',
        post: {
          id: 195,
          location: {
            id: 103,
            country: 'Sierra Leone',
            code: '00A',
            city: 'Freetown',
            state: '',
          },
        },
      },
      status: 'closed',
      draft_date: '2017-12-20',
      submitted_date: '2017-12-20',
      handshake_offered_date: '2017-12-20',
      handshake_accepted_date: '2017-12-20',
      handshake_declined_date: '2017-12-20',
      in_panel_date: '2017-12-20',
      scheduled_panel_date: '2017-12-20',
      approved_date: '2017-12-20',
      declined_date: '2017-12-20',
      closed_date: '2017-12-20',
      create_date: '2017-12-20',
      update_date: '2017-12-20',
      is_priority: false,
      reviewer: {
        username: 'woodwardw',
        first_name: 'Wendy',
        last_name: 'Woodward',
        email: 'woodwardw@state.gov',
        phone_number: '555-555-5555',
        is_cdo: false,
      },
    },
  ],
};

export default bidListObject;
