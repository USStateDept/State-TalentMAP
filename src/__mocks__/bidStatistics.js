const bidStatistics = {
  id: 1,
  name: 'Demo BidCycle 2018-01-10 15:52:20.583434',
  cycle_start_date: '2017-10-10T00:00:00Z',
  cycle_deadline_date: '2018-02-10T00:00:00Z',
  cycle_end_date: '2018-04-10T00:00:00Z',
  total_positions: 251,
  available_positions: 240,
  available_domestic_positions: 81,
  available_international_positions: 159,
  total_bids: 40,
  total_bidders: 4,
  in_panel_bidders: 0,
  approved_bidders: 1,
  bidding_days_remaining: 30,
};

export const bidStatisticsList = {
  results: [
    bidStatistics,
  ],
};

export default bidStatistics;
