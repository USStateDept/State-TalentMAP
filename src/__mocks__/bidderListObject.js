const bidderListObject = {
  count: 6,
  next: null,
  previous: null,
  results: [
    {
      id: 5,
      current_assignment: '(active) Kara Batisak at [58312265] PROJECT DIRECTOR (Guatemala, Guatemala)',
      skills: [
        {
          id: 63,
          cone: null,
          code: '6218',
          description: 'CONSTRUCTION ENGINEERING',
        },
      ],
      grade: '01',
      is_cdo: false,
      primary_nationality: 'United States',
      secondary_nationality: null,
      bid_statistics: [
        {
          id: 1,
          bidcycle: 'Demo BidCycle 2018-01-10 15:52:20.583434',
          user: 'Kara Batisak',
          draft: 6,
          submitted: 3,
          handshake_offered: 1,
          handshake_accepted: 0,
          handshake_declined: 0,
          in_panel: 0,
          approved: 0,
          declined: 0,
          closed: 0,
        },
      ],
      user: {
        username: 'batisak',
        email: 'batisak@state.gov',
        first_name: 'Kara',
        last_name: 'Batisak',
      },
      language_qualifications: [],
    },
    {
      id: 3,
      current_assignment: '(active) John Doe at [10181012] INTERNATIONAL RELATIONS OFF (Singapore, Singapore)',
      skills: [
        {
          id: 55,
          cone: null,
          code: '6080',
          description: 'INTERFUNCTIONAL OFFICER OPER',
        },
      ],
      grade: '02',
      is_cdo: false,
      primary_nationality: 'United States',
      secondary_nationality: null,
      bid_statistics: [],
      user: {
        username: 'doej',
        email: 'doej@talentmap.us',
        first_name: 'John',
        last_name: 'Doe',
      },
      language_qualifications: [],
    },
    {
      id: 1,
      current_assignment: '(active) Guest McGuestson at [58312870] CONSTRUCTION MANAGER (Nogales, Mexico)',
      skills: [
        {
          id: 63,
          cone: null,
          code: '6218',
          description: 'CONSTRUCTION ENGINEERING',
        },
      ],
      grade: '03',
      is_cdo: false,
      primary_nationality: 'United States',
      secondary_nationality: null,
      bid_statistics: [],
      user: {
        username: 'guest',
        email: 'guest@state.gov',
        first_name: 'Guest',
        last_name: 'McGuestson',
      },
      language_qualifications: [],
    },
    {
      id: 6,
      current_assignment: '(active) Tarek Rehman at [56457009] ASST REGIONAL SECURITY OFCR (Adana, Turkey)',
      skills: [
        {
          id: 20,
          cone: null,
          code: '2501',
          description: 'SECURITY',
        },
      ],
      grade: '04',
      is_cdo: false,
      primary_nationality: 'United States',
      secondary_nationality: null,
      bid_statistics: [
        {
          id: 2,
          bidcycle: 'Demo BidCycle 2018-01-10 15:52:20.583434',
          user: 'Tarek Rehman',
          draft: 5,
          submitted: 2,
          handshake_offered: 3,
          handshake_accepted: 0,
          handshake_declined: 0,
          in_panel: 0,
          approved: 0,
          declined: 0,
          closed: 0,
        },
      ],
      user: {
        username: 'rehmant',
        email: 'rehmant@state.gov',
        first_name: 'Tarek',
        last_name: 'Rehman',
      },
      language_qualifications: [],
    },
    {
      id: 2,
      current_assignment: '(active) Administrator TalentMAP at [58312869] PROJECT DIRECTOR (Nogales, Mexico)',
      skills: [
        {
          id: 63,
          cone: null,
          code: '6218',
          description: 'CONSTRUCTION ENGINEERING',
        },
      ],
      grade: '01',
      is_cdo: false,
      primary_nationality: 'United States',
      secondary_nationality: null,
      bid_statistics: [],
      user: {
        username: 'admin',
        email: 'admin@talentmap.us',
        first_name: 'Administrator',
        last_name: 'TalentMAP',
      },
      language_qualifications: [],
    },
    {
      id: 4,
      current_assignment: '(active) Jenny Townpost at [10026129] INTERNATIONAL RELATIONS OFF (Port Moresby, Papua New Guinea)',
      skills: [
        {
          id: 55,
          cone: null,
          code: '6080',
          description: 'INTERFUNCTIONAL OFFICER OPER',
        },
      ],
      grade: '02',
      is_cdo: false,
      primary_nationality: 'United States',
      secondary_nationality: null,
      bid_statistics: [
        {
          id: 4,
          bidcycle: 'Demo BidCycle 2018-01-10 15:52:20.583434',
          user: 'Jenny Townpost',
          draft: 3,
          submitted: 4,
          handshake_offered: 3,
          handshake_accepted: 0,
          handshake_declined: 0,
          in_panel: 0,
          approved: 0,
          declined: 0,
          closed: 0,
        },
      ],
      user: {
        username: 'townpostj',
        email: 'townpostj@state.gov',
        first_name: 'Jenny',
        last_name: 'Townpost',
      },
      language_qualifications: [],
    },
  ],
};

export default bidderListObject;
