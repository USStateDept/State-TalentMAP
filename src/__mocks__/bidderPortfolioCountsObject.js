const bidderPortfolioCountsObject = {
  all_clients: 100,
  bidding_clients: 50,
  in_panel_clients: 20,
  on_post_clients: 30,
  bidding_no_handshake: 10,
};

export default bidderPortfolioCountsObject;
