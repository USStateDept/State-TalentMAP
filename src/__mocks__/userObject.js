export const bidderUserObject = {
  skill_code: ['INFORMATION MANAGEMENT (2880)', 'ADMINISTRATIVE SUPPORT (2080)'],
  user: {
    first_name: 'John',
    last_name: 'Doe',
    email: 'doej@state.gov',
    username: 'jdoe',
  },
  is_cdo: false,
  cdo: {
    email: 'shadtrachl@state.gov',
  },
  bid_statistics: [
    {
      draft: 1,
      submitted: 1,
    },
  ],
};

export const cdoUserObject = Object.assign({}, bidderUserObject, { is_cdo: true });
