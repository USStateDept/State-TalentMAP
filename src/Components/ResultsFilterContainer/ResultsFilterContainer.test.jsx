import { shallow } from 'enzyme';
import React from 'react';
import toJSON from 'enzyme-to-json';
import sinon from 'sinon';
import ResultsFilterContainer from './ResultsFilterContainer';
import { bidderUserObject } from '../../__mocks__/userObject';

describe('ResultsFilterContainerComponent', () => {
  const items = [{
    title: 'title', expanded: true,
  }];

  const props = {
    filters: items,
    onQueryParamUpdate: () => {},
    onChildToggle: () => {},
    onQueryParamToggle: () => {},
    resetFilters: () => {},
    setAccordion: () => {},
    fetchMissionAutocomplete: () => {},
    missionSearchResults: [],
    missionSearchIsLoading: false,
    missionSearchHasErrored: false,
    fetchPostAutocomplete: () => {},
    postSearchResults: [],
    postSearchIsLoading: false,
    postSearchHasErrored: false,
    userProfile: bidderUserObject,
  };

  it('is defined', () => {
    const wrapper = shallow(
      <ResultsFilterContainer
        {...props}
      />);
    expect(wrapper).toBeDefined();
  });

  it('can receive props', () => {
    const wrapper = shallow(<ResultsFilterContainer
      {...props}
    />);
    expect(wrapper.instance().props.filters).toBe(items);
  });

  it('is defined after receiving new props', () => {
    const wrapper = shallow(<ResultsFilterContainer
      {...props}
    />);
    wrapper.setProps({ newProp: 'abz' });
    expect(wrapper).toBeDefined();
  });

  it('can call the onQueryParamUpdate function', () => {
    const value = 1;
    const queryUpdateSpy = sinon.spy();
    const toggleSpy = sinon.spy();
    const wrapper = shallow(<ResultsFilterContainer
      {...props}
      onQueryParamUpdate={queryUpdateSpy}
      onChildToggle={toggleSpy}
    />);
    wrapper.instance().onQueryParamUpdate(value);
    sinon.assert.calledOnce(queryUpdateSpy);
    sinon.assert.calledOnce(toggleSpy);
  });

  it('can call the onQueryParamToggle function', () => {
    const value = 1;
    const queryToggleSpy = sinon.spy();
    const toggleSpy = sinon.spy();
    const wrapper = shallow(<ResultsFilterContainer
      {...props}
      onChildToggle={toggleSpy}
      onQueryParamToggle={queryToggleSpy}
    />);
    wrapper.instance().onQueryParamToggle(value, value, value);
    sinon.assert.calledOnce(queryToggleSpy);
    sinon.assert.calledOnce(toggleSpy);
  });

  it('matches snapshot', () => {
    const wrapper = shallow(<ResultsFilterContainer
      {...props}
    />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
