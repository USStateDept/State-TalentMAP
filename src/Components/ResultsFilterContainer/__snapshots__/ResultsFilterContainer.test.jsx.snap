// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultsFilterContainerComponent matches snapshot 1`] = `
<div
  className="filter-container"
>
  <div
    className="filter-container-bottom"
  >
    <div
      className="usa-grid-full filter-control-container"
    >
      <div
        className="filter-control-left"
      >
        Select Filter:
      </div>
      <div
        className="filter-control-right"
      >
        <ResetFilters
          resetFilters={[Function]}
        />
      </div>
    </div>
    <div
      className="usa-grid-full search-filters-container"
    >
      <SearchFiltersContainer
        fetchMissionAutocomplete={[Function]}
        fetchPostAutocomplete={[Function]}
        filters={
          Array [
            Object {
              "expanded": true,
              "title": "title",
            },
          ]
        }
        isCDO={false}
        missionSearchHasErrored={false}
        missionSearchIsLoading={false}
        missionSearchResults={Array []}
        postSearchHasErrored={false}
        postSearchIsLoading={false}
        postSearchResults={Array []}
        queryParamToggle={[Function]}
        queryParamUpdate={[Function]}
        selectedAccordion={
          Object {
            "main": "",
            "sub": "",
          }
        }
        setAccordion={[Function]}
      />
    </div>
  </div>
</div>
`;
